lib/spack/env/cc

```diff
@@ -339,7 +339,15 @@ case "$mode" in
         args=("${args[@]}" ${SPACK_LDLIBS[@]}) ;;
 esac
 
-full_command=("$command" "${args[@]}")
+#ccache only supports C languages, so filtering out Fortran
+if [[ ${lang_flags} != "F" && ${SPACK_CCACHE_BINARY} ]]; then
+    full_command=("${SPACK_CCACHE_BINARY}" "$command" "${args[@]}")
+    # #3761#issuecomment-294352232
+    # workaround for stage being a temp folder
+    export CCACHE_NOHASHDIR=yes
```

Comment:
Sure, if we would compile all versions of a package in same directory, otherwise there will no compile re-usage. As @citibeth said above as long as spack removes the source, there is no real point to worry about that issue.